pkg/brokerapi/openservicebroker/open_service_broker_client.go

```diff
@@ -140,16 +135,17 @@ func (c *openServiceBrokerClient) GetCatalog() (*brokerapi.Catalog, error) {
 }
 
 func (c *openServiceBrokerClient) CreateServiceInstance(ID string, req *brokerapi.CreateServiceInstanceRequest) (*brokerapi.CreateServiceInstanceResponse, int, error) {
-	var serviceInstanceURL string
-
-	if req.AcceptsIncomplete {
-		serviceInstanceURL = fmt.Sprintf(serviceInstanceAsyncFormatString, c.url, ID)
-	} else {
-		serviceInstanceURL = fmt.Sprintf(serviceInstanceFormatString, c.url, ID)
-	}
-
-	// TODO: Handle the auth
-	resp, err := sendOSBRequest(c, http.MethodPut, serviceInstanceURL, req)
+	serviceInstanceURL := fmt.Sprintf(serviceInstanceFormatString, c.url, ID)
+
+	resp, err := sendOSBRequest(
+		c,
+		http.MethodPut,
+		serviceInstanceURL,
+		map[string]string{
+			"accepts_incomplete": fmt.Sprintf("%t", req.AcceptsIncomplete),
```

Comment:
Would be kind of nice if we didn't send the query param at all in the 'false' case but I guess we could do that later.